skule_vote/backend/tests.py

```diff
@@ -642,3 +645,377 @@ def test_yes_cookie_returns_eligible(self):
         self.assertJSONEqual(
             str(response.content, encoding="utf8"), {"voter_eligible": True}
         )
+
+
+class BallotSubmitViewTestCase(SetupMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.ballot_submit_view = reverse("api:backend:ballot-submit")
+        self.cookie_view = reverse("api:backend:bypass-cookie")
+
+        # Empty and started election session
+        self.election_session = self._create_election_session(
+            self._set_election_session_data()
+        )
+
+    def test_no_cookie_permission_denied(self):
+        response = self.client.get(self.ballot_submit_view)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_invalid_cookie_permission_denied(self):
+        voter_dict = self._urlencode_cookie_request()
+
+        response = self.client.post(self.cookie_view, voter_dict, follow=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            self.client.cookies["student_number_hash"].value.split(":")[0],
+            voter_dict["pid"],
+        )
+
+        hash_signature = "".join(
+            self.client.cookies["student_number_hash"].value.split(":")[1:]
+        )
+        self.client.cookies["student_number_hash"].set(
+            "student_number_hash",
+            "im hacking the hash:" + hash_signature,
+            "im hacking the hash:" + hash_signature,
+        )
+
+        response = self.client.get(self.ballot_submit_view)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_invalid_election_id(self):
+        voter_dict = self._urlencode_cookie_request()
+        response = self.client.post(self.cookie_view, voter_dict, follow=True)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Invalid election Id
+        payload = {"electionId": "doesn't exist", "ranking": {}}
```

Comment:
for the payload where it should be an id or a number always I'd recommend using a number otherwise you're adding a failure mode in conversion